datafusion/functions/src/math/random.rs

```diff
@@ -69,12 +69,13 @@ impl ScalarUDFImpl for RandomFunc {
     }
 }
 
-/// Random SQL function
 fn random(args: &[ColumnarValue]) -> Result<ColumnarValue> {
-    let len: usize = match &args[0] {
-        ColumnarValue::Array(array) => array.len(),
-        _ => return exec_err!("Expect random function to take no param"),
+    let len = if args.is_empty() {
+        1
+    } else {
+        return exec_err!("Expect random function to take no param");
```

Comment:
This might be more consistent with the other changes in this PR:

```suggestion
        return exec_err!("Expect {} function to take no parameters", self.name());
```

Reply:
I did this before, and I noticed that `random` is used in test so I need to move it out as an independent function, but adding `name` as another argument looks so weird, so I keep it like this.

Reply:
we can even return ScalarValue::F64 instead of Array. Let me fix these too